sklearn/svm/tests/test_svm.py

```diff
@@ -1398,23 +1397,21 @@ def test_linearsvm_liblinear_sample_weight(SVM, params):
             assert_allclose(X_est_no_weight, X_est_with_weight)
 
 
-def test_n_support_oneclass_svr():
+@pytest.mark.parametrize("Klass", (OneClassSVM, SVR, NuSVR))
+def test_n_support(Klass):
     # Make n_support is correct for oneclass and SVR (used to be
     # non-initialized)
     # this is a non regression test for issue #14774
     X = np.array([[0], [0.44], [0.45], [0.46], [1]])
-    clf = svm.OneClassSVM()
-    assert not hasattr(clf, "n_support_")
-    clf.fit(X)
-    assert clf.n_support_ == clf.support_vectors_.shape[0]
-    assert clf.n_support_.size == 1
-    assert clf.n_support_ == 3
-
     y = np.arange(X.shape[0])
-    reg = svm.SVR().fit(X, y)
-    assert reg.n_support_ == reg.support_vectors_.shape[0]
-    assert reg.n_support_.size == 1
-    assert reg.n_support_ == 4
+    est = Klass()
+    fit_data = {"X": X}
+    if Klass.__name__ != "OneClassSVM":
+        fit_data.update({"y": y})
+    assert not hasattr(est, "n_support_")
+    est.fit(**fit_data)
```

Comment:
Since `OneClassSVM` ignores `y`, I am okay with calling `est.fit(X, y)` directly to reduce the complexity in this test.